pkg/retry/retry.go

```diff
@@ -125,6 +125,12 @@ func (s *Server) handleRetry() http.Handler {
 				return
 			}
 
+			if len(deliveries) == 0 {
```

Comment:
Its possible to reach line 137 and experience a NPE. This change should prevent this case.